deployments/constructs/istio.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2023 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository https://github.com/carbynestack/carbynestack.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Construct } from "constructs";
+import * as cdktf from "cdktf";
+import * as kubernetes from "@cdktf/provider-kubernetes";
+import * as helm from "@cdktf/provider-helm";
+
+export interface IstioConfig {
+  idPostfix?: string;
+  dependsOn: cdktf.ITerraformDependable[];
+  helmProvider?: cdktf.TerraformProvider;
+  kubernetesProvider?: cdktf.TerraformProvider;
+}
+
+export class Istio extends Construct {
+  public ingressIP: string;
+  public istioIngressGatewayService: kubernetes.dataKubernetesService.DataKubernetesService;
+
+  constructor(scope: Construct, name: string, config: IstioConfig) {
+    super(scope, name);
+
+    const istioNamespace = new kubernetes.namespace.Namespace(
+      this,
+      `istio-system${config.idPostfix}`,
+      {
+        provider: config.kubernetesProvider,
+        metadata: {
+          name: "istio-system",
+        },
+      },
+    );
+
+    const istioBase = new helm.release.Release(
+      this,
+      `istio-base${config.idPostfix}`,
+      {
+        dependsOn: [...config.dependsOn],
+        wait: true,
+        waitForJobs: true,
+        provider: config.helmProvider,
+        name: "istio-base",
+        chart: "base",
+        namespace: istioNamespace.metadata.name,
+        repository: "https://istio-release.storage.googleapis.com/charts",
+      },
+    );
+
+    // istio control plane - https://istio.io/latest/blog/2020/istiod/
+    const istioD = new helm.release.Release(this, `istiod${config.idPostfix}`, {
+      dependsOn: [...config.dependsOn, istioBase],
+      provider: config.helmProvider,
+      name: "istiod",
+      chart: "istiod",
+      namespace: istioNamespace.metadata.name,
+      repository: "https://istio-release.storage.googleapis.com/charts",
+      wait: true,
+      waitForJobs: true,
+    });
+
+    // istio ingress
+
+    const istioIngressGatewayPorts = [
+      { name: "status-port", port: "15021", targetPort: "15021" },
+      { name: "http2", port: "80", targetPort: "8080" },
+      { name: "https", port: "443", targetPort: "8443" },
+      { name: "tcp", port: "31400", targetPort: "31400" },
+      { name: "tls", port: "15443", targetPort: "15443" },
+      { name: "ephemeral-mpc-engine-port-0", port: "30000" },
```

Comment:
It should be configurable by the user which (or how many) ports to open.
Should be fine for now (as it is the same as for the manual deployment), but I'd expect this to be also generated by a loop in the long run)
[consider this comment more to be something like a note] ;-)